oi/type_graph/NodeTracker.h

```diff
@@ -109,23 +109,25 @@ class NodeTrackerHolder {
 };
 
 /*
- * MutationTracker
+ * ResultTracker
  *
- * Helper class for mutators. Efficiently tracks visited and replaces nodes.
+ * Efficiently caches the results of visited nodes.
  */
-class MutationTracker {
+template <typename T>
+class ResultTracker {
  public:
-  MutationTracker(size_t size) : visited_(size) {
+  ResultTracker() = default;
+  ResultTracker(size_t size) : visited_(size) {
   }
 
   /*
    * get
    *
-   * Returns a type pointer if the given node has been visited or replaced.
+   * Returns the cached result if the given node has been visited.
    * Returns nullptr if this node has not yet been seen.
```

Comment:
s/nullptr/a default constructed element/?

Reply:
I hadn't actually changed enough code to handle arbitrary types - I've added a static_assert that the result type is a pointer now. I'll leave handling arbitrary types as a future extension.